lib/compile-exports.js

```diff
@@ -17,21 +17,31 @@ module.exports = function compileExports(result, importItemMatcher, camelCaseKey
     function addEntry(k) {
       res.push("\t" + JSON.stringify(k) + ": " + valueAsString);
     }
-    if (camelCaseKeys !== 'only' && camelCaseKeys !== 'dashesOnly') {
-      addEntry(key);
-    }
 
-    var targetKey;
-    if (camelCaseKeys === true || camelCaseKeys === 'only') {
-      targetKey = camelCase(key);
-      if (targetKey !== key) {
```

Comment:
this line was the culprit, as it would prevent the camelcased key to be added to the locals only if it was different from the default key. However with the `only` and `dashesOnly` options the default key would not have been added -> empty locals.